core/src/main/scala/kafka/common/ZkNodeChangeNotificationListener.scala

```diff
@@ -58,6 +59,7 @@ class ZkNodeChangeNotificationListener(private val zkUtils: ZkUtils,
   def init() {
     zkUtils.makeSurePersistentPathExists(seqNodeRoot)
     zkUtils.zkClient.subscribeChildChanges(seqNodeRoot, NodeChangeListener)
+    zkUtils.zkClient.subscribeStateChanges(ZkStateChangeListener)
     processAllNotifications()
```

Comment:
Aren't we calling processAllNotifications upon session establishment in the ZkStateChangeLIstener? I'm wondering if calling it here makes sense as the session might not have been established yet.